internal/kuberesource/mutators.go

```diff
@@ -298,33 +324,73 @@ func MapPodSpecWithMeta(
 		meta *applymetav1.ObjectMetaApplyConfiguration,
 		spec *applycorev1.PodSpecApplyConfiguration,
 	) (*applymetav1.ObjectMetaApplyConfiguration, *applycorev1.PodSpecApplyConfiguration),
-) any {
+) (any, error) {
```

Comment:
Did you consider skipping the invalid cases instead of returning an error?

Reply:
It turns out, genpolicy also checks these fields, so we could just ignore invalid cases. The program would not panic and genpolicy will show a reasonable error message with `missing field 'spec'` and a path like `spec.template.spec`.